packages/solidity-contracts/contracts/fuelchain/v2/FuelMessagePortalV2.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.9;
+
+import '../FuelMessagePortal.sol';
+
+contract FuelMessagePortalV2 is FuelMessagePortal {
+
+    error AccountDepositLimit();
+    error GlobalDepositLimit();
+
+    uint public depositLimitGlobal;
+    uint public depositLimitPerAccount;
+
+    uint public totalDeposited;
+    mapping(address => uint) public depositedAmounts;
+
+
+    function initializeV2(
```

Comment:
> more trust because we cannot change things later for whatever reason

Contract being upgradable defeats that. As for the rest of the points, I think it might have some issues with our current CI / CD process. Tagging @luizstacio : you were having problems with immutables / constants, right?

Reply:
Solved by making limits immutables